### PR TITLE
Autogenerate TTNN Tests for Explorer CI to consume

### PR DESCRIPTION
**Pull request by vprajapati-tt**
- Leverage `check-ttmlir` target to autogenerate tests to visualize.
- Depends on #1298

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -504,8 +504,23 @@ jobs:
...
+        source env/activate
+        cmake --build ${{ steps.strings.outputs.build-output-dir }} -- ttrt
+
+    - name: Generate tests
+      shell: bash
+      run: |
+        source env/activate
+        ttrt query --save-artifacts
+        export SYSTEM_DESC_PATH="${GITHUB_WORKSPACE}/ttrt-artifacts/system_desc.ttsys"
+        cmake --build ${{ steps.strings.outputs.build-output-dir }} -- check-ttmlir
```
Comment (odjuricicTT) [inline .github/workflows/build-and-test.yml:519]: Hmm, could this cause a bit of confusion in the CI since we are rerunning the same tests from another job? If someone makes a mistake in FileCheck they will see the error in explorer job as well, which might not be what we want.

Comment (odjuricicTT) [reply .github/workflows/build-and-test.yml:519]: Not sure about this one. @nobradovictt Thoughts?

**Review by odjuricicTT — COMMENTED** [review]
Great to have this functionality so we can run explorer no all these files. Not sure if we want it in CI tho. Left a comment.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -504,8 +504,23 @@ jobs:
         source env/activate
         cmake --build ${{ steps.strings.outputs.build-output-dir }} -- explorer
 
+    - name: Build ttrt
```
[the remark this answers is not in the document]
Comment (vprajapati-tt) [reply .github/workflows/build-and-test.yml:507]: Alright, I'll clean it up. Thanks for the note.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (nsmithtt) [reply .github/workflows/build-and-test.yml:519]: Maybe we could have a dedicated subdir for explorer tests that are still underneath the `/tests` root dir?

```bash
llvm-lit build/test/ttmlir/Explorer
```

Comment (odjuricicTT) [reply .github/workflows/build-and-test.yml:519]: @vprajapati-tt suggested that we move explorer tests to `/tests` and we agreed that makes sense.

Comment (vprajapati-tt) [reply .github/workflows/build-and-test.yml:519]: My thoughts on this are:

We should have explorer specific tests, and it would be best if they reside in `/tests/` so we get coverage on if dialect changes are incompatible with the tests that we've written.

It's also important for us to have coverage for the explorer parsing as well. While our tests can test the capability of the tool and target edge cases, the tool is meant to be generalized for `tt-mlir` dialects. As such, holding increased coverage from these tests will provide exposure to the flaws of how models are being parsed in explorer (ex: a NamedAttribute called `shape` can have different underlying types, and this was causing a problem with the llama tests).

I think there is much to gain with autogenerating since even unrelated changes can cause problems in the explorer parsing engine, and it would be best for us to notice that without having to manually test these cases / write them in.

Comment (odjuricicTT) [reply .github/workflows/build-and-test.yml:519]: Ok, let's try this out provided we do the following:
1. We don't use check-ttmlir within the explorer job in order to not have explorer CI red when FileCheck is failing.
2. The job is not the slowest one on CI.

File: tools/explorer/CMakeLists.txt [hunk 3]
```diff
@@ -17,7 +17,7 @@ ExternalProject_Add(
 
 add_custom_target(explorer
   COMMENT "Building tt-explorer... ${TTMLIR_BIN_DIR}"
-  COMMAND pip install ${CMAKE_CURRENT_SOURCE_DIR}/tt_adapter
+  COMMAND pip install $<$<CONFIG:Debug>:-e> ${CMAKE_CURRENT_SOURCE_DIR}/tt_adapter
```
Comment (odjuricicTT) [inline tools/explorer/CMakeLists.txt:20]: Can we introduce a new flag for this? The idea being to keep the build process consistent in all CI builds, al also for people not trying to debug explorer.

**Review by vmilosevic — APPROVED** [review]

**Review by tapspatel — APPROVED** [review]

**Review by odjuricicTT — CHANGES_REQUESTED** [review]
Once we get the tests to pass, let's also check the duration of the job before merging this.

**Comment by vprajapati-tt** [comment]
Awesome. we're finally seeing some additional coverage and catching failpoints in explorer :) I'll address these fixes + add TTIR modules from the `tests` directory.

**Comment by vprajapati-tt** [comment]
> Once we get the tests to pass, let's also check the duration of the job before merging this.

6m 19s. Seems pretty reasonable. Let me know what you think, I'd love to get more coverage for explorer soon :)

File: tools/explorer/test/run_tests.py [hunk 4]
```diff
@@ -8,26 +8,61 @@
...
+
+
+def get_test_files(paths):
+    files = []
+    for path in paths:
+        files.extend(glob.glob(path, recursive=True))
+    return files
+
+
+def execute_command(model_path, settings):
```
Comment (odjuricicTT) [inline tools/explorer/test/run_tests.py:39]: This already exits as well.

**Review by odjuricicTT — APPROVED** [review]

File: tools/explorer/test/run_tests.py (hunk 4 above)
Comment (vprajapati-tt) [reply tools/explorer/test/run_tests.py:39]: Yeah I don't think I added these, git diff seems weird for some reason.
